File: bank-service/src/main/java/rs/edu/raf/mapper/CurrencyMapper.java
package rs.edu.raf.mapper;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;
import rs.edu.raf.dto.CurrencyCreateDTO;
import rs.edu.raf.dto.CurrencyDTO;
import rs.edu.raf.model.Currency;
import rs.edu.raf.repository.CountryRepository;

@Component
@AllArgsConstructor
public class CurrencyMapper {
    private CountryMapper countryMapper;
    private CountryRepository countryRepository;

    public Currency currencyCreateDTOtoCurrency(CurrencyCreateDTO currencyCreateDTO) {
        Currency currency = new Currency();

        currency.setCode(currencyCreateDTO.code());
        currency.setName(currencyCreateDTO.name());
        currency.setSymbol(currencyCreateDTO.symbol());
        currency.setCountry(countryRepository.findById(currencyCreateDTO.countryId()).orElseThrow());

        return currency;
    }
    public CurrencyDTO currencyToCurrencyDTO(Currency currency) {
        return new CurrencyDTO(currency.getId(), currency.getName(), currency.getCode(), currency.getSymbol(),
                countryMapper.countryToCountryDTO(currency.getCountry()));
    }
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/mapper/CurrencyMapper.kt
package org.example.bankservicekotlin.mapper

import org.example.bankservicekotlin.dto.CurrencyDTO
import org.example.bankservicekotlin.dto.CurrencyCreateDTO
import org.example.bankservicekotlin.model.Currency
import org.example.bankservicekotlin.repository.CountryRepository
import org.springframework.stereotype.Component

@Component
class CurrencyMapper(private val countryMapper: CountryMapper, private val countryRepository: CountryRepository) {
    fun currencyCreateDTOtoCurrency(currencyCreateDTO: CurrencyCreateDTO): Currency {
        val country = countryRepository.findById(currencyCreateDTO.countryId).orElseThrow()
        return Currency(null, currencyCreateDTO.name, currencyCreateDTO.code, currencyCreateDTO.symbol, country)
    }

    fun currencyToCurrencyDTO(currency: Currency): CurrencyDTO {
        return CurrencyDTO(
            currency.id,
            currency.name,
            currency.code,
            currency.symbol,
            countryMapper.countryToCountryDTO(currency.country)
        )
    }
}